### PR TITLE
[MM-52829] Migrate "channels/src/components/admin_console/team_channel_settings/details/index.js" to TypeScript

### PR DESCRIPTION
**Pull request by ridwankabeer435**
…/team/details/index file to TypeScript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/23431
Jira https://mattermost.atlassian.net/browse/MM-52829

#### Release Note

```release-note

```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 1]
```diff
@@ -17,16 +17,19 @@ import {
...
-    const totalGroups = groups.length;
-    const isLicensedForLDAPGroups = state.entities.general.license.LDAPGroups === 'true';
+import TeamDetails, {Props as TeamDetailsProps} from './team_details';
+import {GlobalState} from 'types/store';
+import {Team} from '@mattermost/types/teams';
+import {Group} from '@mattermost/types/groups';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+
+function mapStateToProps(state: GlobalState, props: TeamDetailsProps) {
+    const teamID: string = props.teamID;
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:27]: why was this changed from `const teamID = props.match.params.team_id;` ?

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 2]
```diff
@@ -17,16 +17,19 @@ import {
...
+    const team: Team = getTeam(state, teamID);
+    const groups: Group[] = getGroupsAssociatedToTeam(state, teamID);
+    const allGroups: Record<string, Group> = getAllGroups(state);
+    const totalGroups: number = groups.length;
+    const isLicensedForLDAPGroups: boolean = state.entities.general.license.LDAPGroups === 'true';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:32]: lets remove this explicit typing since they the types are returned from the selectors themselves.

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 3]
```diff
@@ -57,4 +60,4 @@ function mapDispatchToProps(dispatch) {
     };
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(TeamDetails);
+export default connect(mapStateToProps, mapDispatchToProps)(TeamDetails);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:63]: nit : extra line at the end of file.

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 4]
```diff
@@ -17,16 +17,19 @@ import {
...
-    const team = getTeam(state, teamID);
-    const groups = getGroupsAssociatedToTeam(state, teamID);
-    const allGroups = getAllGroups(state, teamID);
-    const totalGroups = groups.length;
-    const isLicensedForLDAPGroups = state.entities.general.license.LDAPGroups === 'true';
+import TeamDetails, {Props as TeamDetailsProps} from './team_details';
+import {GlobalState} from 'types/store';
+import {Team} from '@mattermost/types/teams';
+import {Group} from '@mattermost/types/groups';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:24]: nit: imports rearrangements

- library imports
space
- imports from mattermost/types
space
- imports from mattermost-redux
space
- others
space
import TeamDetails, {Props as TeamDetailsProps} from './team_details';

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by M-ZubairAhmed** [comment]
@ridwankabeer435 please check the failing CI

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts (hunk 1 above)
Comment (ridwankabeer435) [reply webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:27]: `props.match.params` is not recognized by TypeScript's transpiler after importing the `props` as `TeamDetailsProps` from the `TeamDetails` component. TypeScript's transpiler also does not recognize `props` without explicitly importing them from `TeamDetails`. Using `props.teamID` resolves the above issue.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:27]: In the original code, `props.match.params.team_id` was used to extract the `team_id` parameter from the match object within the props. This assumes that the component is being rendered within a routing context where the `team_id` parameter is available as part of the URL.

Comment (ridwankabeer435) [reply webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts]: Reverted the code to use `props.match.params.team_id` to obtain team_jd and added an `OwnProps `object to replace `TeamDetailsProps` to pass as parameter

**Review by harshilsharma63 — APPROVED** [review]

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 5]
```diff
@@ -15,16 +16,25 @@ import {
...
-import TeamDetails from './team_details';
+import TeamDetails, {Props as TeamDetailsProps} from './team_details';
+
+type OwnProps = {
+    match: {
+        params: {
+            teamID: string;
+        };
+    };
+}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:31]: You can make use of types from react-router like this

https://github.com/mattermost/mattermost/blob/c943ed6859b1be013e4da92b1952c607a35258bc/webapp/channels/src/components/channel_layout/center_channel/index.ts#L26-L30

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 6]
```diff
@@ -1,12 +1,13 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators} from 'redux';
+import {ActionCreatorsMapObject, Dispatch, bindActionCreators} from 'redux';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts:5]: nit : remove this space

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts [hunk 7]
```diff
@@ -15,16 +16,25 @@ import {
...
+        params: {
+            teamID: string;
+        };
+    };
+}
 
-function mapStateToProps(state, props) {
-    const teamID = props.match.params.team_id;
+function mapStateToProps(state: GlobalState, props: OwnProps) {
+    const teamID = props.match.params.teamID;
```
[the remark this answers is not in the document]
Comment (ridwankabeer435) [reply webapp/channels/src/components/admin_console/team_channel_settings/team/details/index.ts]: Reverted to `const teamID = props.match.params.team_id`

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by yasserfaraazkhan — APPROVED** [review]
